src/test/java/teammates/test/MockLogsProcessor.java

```diff
@@ -105,4 +117,129 @@ public void createFeedbackSessionLog(String courseId, String email, String fsNam
+                        && logEntry.getUserInfo().regkey.equals(regkey);
+            }
+            if (email != null) {
+                matchEmail = logEntry.getUserInfo() != null && logEntry.getUserInfo().email != null
+                        && logEntry.getUserInfo().email.equals(email);
+            }
+            if (logEvent != null) {
+                matchLogEvent = logEntry.getLogEvent() != null && logEntry.getLogEvent().equals(logEvent);
+            }
+            if (sourceLocation != null && sourceLocation.getFile() != null) {
+                matchSourceLocation = logEntry.getSourceLocation() != null
+                        && logEntry.getSourceLocation().getFile() != null
+                        && logEntry.getSourceLocation().getFile().equals(sourceLocation.getFile());
+                if (sourceLocation.getFunction() != null) {
+                    matchSourceLocation = matchSourceLocation && logEntry.getSourceLocation().getFunction() != null
+                            && logEntry.getSourceLocation().getFunction().equals(sourceLocation.getFunction());
+                }
+            }
+            if (exceptionClass != null) {
+                matchExceptionClass = logEntry.getExceptionClass() != null
+                        && logEntry.getExceptionClass().equals(exceptionClass);
+            }
+            return matchSeverity && matchTimePeriod && matchTrace && matchApiEndpoint && matchGoogleId && matchRegkey
+                    && matchEmail && matchLogEvent && matchSourceLocation && matchExceptionClass;
+        };
+    }
+
+    /**
+     * Allows mocking of {@link GeneralLogEntry}.
+     */
+    public static class MockGeneralLogEntry extends GeneralLogEntry {
+        private final String apiEndpoint;
+        private final UserInfo userInfo;
+        private final String logEvent;
+        private final String exceptionClass;
+
+        public MockGeneralLogEntry(String logName, String severity, String trace, SourceLocation sourceLocation,
+                long timestamp, String apiEndpoint, String googleId, String regkey, String email, String logEvent,
+                String exceptionClass) {
+            super(logName, severity, trace, sourceLocation, timestamp);
+            this.apiEndpoint = apiEndpoint;
+            this.userInfo = new UserInfo(googleId, regkey, email);
+            this.logEvent = logEvent;
+            this.exceptionClass = exceptionClass;
+        }
+
+        public String getApiEndpoint() {
+            return apiEndpoint;
+        }
+
+        public UserInfo getUserInfo() {
+            return userInfo;
+        }
+
+        public String getLogEvent() {
+            return logEvent;
+        }
+
+        public String getExceptionClass() {
+            return exceptionClass;
+        }
+
+        /**
+         * Represents user information.
+         */
+        public static class UserInfo {
```

Comment:
Does this have to be a mock? This looks like it can be a class usable in production code.

Reply:
It is updated to be under the `datatransfer` package